### ISSUE TITLE
Race condition for graph.view() under Windows 10?

### ISSUE THREAD

**Issue by bjascob** [xflr6/graphviz#107]
I'm using graphviz as part of [amrlib](https://github.com/bjascob/amrlib/blob/master/amrlib/graph_processing/amr_plot.py), which contains a QT based GUI.  One of the buttons on the UI generates a graph and calls graph.view() to generate and display the `png` file.

Under Ubuntu everything works great, but when testing on Windows 10, I notice that the default Windows `png` viewer will open with an empty screen on the first call to graph.view() and just sit there.  If I close that window and push the button that calls graph.view() again, the image will pop-up like it should.  

It looks to me like the file might not be fully written/flushed to disk when the viewer opens which cause the screen to hangout empty until you close it and re-open it again.  Are you familiar with this issue and is there a simple fix to it?

**Comment by xflr6** [comment]
That looks unexpected to me:  `view()` calls `render()` first, which [waits for the rendering process](https://github.com/xflr6/graphviz/blob/21bf42ac043433c157334d9f18057b5c1f739d3c/graphviz/backend.py#L171) ([docs](https://docs.python.org/3/library/subprocess.html#subprocess.Popen.communicate)).

After the rendering process has terminated, I would expect the output file to be fully written and closed.

**Comment by xflr6** [comment]
Perhaps you can test the hypothesis on this being related cache-flushing by trying with the [disk write cache disabled](https://www.maketecheasier.com/disable-disk-write-caching-windows10/) (I would tend to expect if a new process opens a file that has been written not yet synched to disk to block until the sync is done though)?

I haven't seen this behaviour so far e.g. with PDF-viewers and Windows 7. Closing for now until we see this coming up  for others or can reproduce this better SGTM. Maybe first check if playing with the cache settings provides more insight?

**Comment by bjascob** [comment]
Unfortunately this is very intermittent.  It'll happen a few times and then I can't get it to happen again.  I did try turning off the disk cache and I wasn't able to get it to occur.  However, after re-enabling the cache it didn't occur immediately either so there's no cause-effect here.  

This is a very minor issue since it's infrequent and simply re-plotting fixes the issue.  If I see it come up later in additional testing I'll try to dig into it more.  Closing for now.
